src/reference/config/bindings/binding-mqtt-kafka.md

```diff
@@ -102,6 +129,117 @@ Kafka topic used for routing mqtt messages.
 
 Compacted Kafka topic for storing mqtt retained messages.
 
+#### options.clients
+
+> `string`
+
+Pattern defining how to extract client identity from the topic. Using this we can ensure that all messages for the same client identity are produced to Kafka on the same topic partition.
+
+```yaml
+options:
+  clients:
+    - /clients/{identity}/#
+```
+
+### routes
+
+> `array` of `object`
+
+Conditional `mqtt-kafka`-specific routes when adapting `mqtt` topic streams to `kafka` topic streams.
+
+```yaml
+routes:
+  - when:
+      - publish:
+          - topic: /clients/#
+      - subscribe:
+          - topic: /clients/#
+    with:
+      messages: mqtt-clients
+    exit: kafka_cache_client
+  - when:
+      - subscribe:
+          - topic: /sensor-clients/#
+      - subscribe:
+          - topic: /device-clients/#
+    with:
+      messages: mqtt-devices
+    exit: kafka_cache_client
+```
+
+### routes[].guarded
+
+> `object` as named map of `string:string` `array`
+
+Roles required by named guard.
```

Comment:
It would be better to use the same text as the other `guarded` sections unless this works differently

```
List of roles required by each named guard to authorize this route.
```